src/main/java/com/selfxdsd/selfweb/api/ContractsApi.java

```diff
@@ -329,4 +366,51 @@ public ContractsApi(final User user, final Self self) {
         }
         return resp;
     }
+
+    /**
+     * Restores a Contract marked for removal.
+     * @param owner Owner of the project (username or org name).
+     * @param name Simple name of the project.
+     * @param username Contributor's username.
+     * @param role Contributor's role.
+     * @return JsonResponse.
+     * @checkstyle ParameterNumber (10 lines)
+     */
+    @PutMapping(
+        value = "/projects/{owner}/{name}/contracts/{username}/restore",
```

Comment:
@amihaiemil Wouldn't be better to use here `@DeleteMapping` with `/projects/{owner}/{name}/contracts/{username}/mark` and for mark for removal endpoint  `@PutMapping`? This way we'll share the same endpoint.

Reply:
@criske Why use DeleteMapping when this endpoint restores a contract? I think Put is the correct way :-?